ext-src/models/IqComponentModel.ts

```diff
@@ -50,62 +50,79 @@ export class IqComponentModel implements ComponentModel {
       return new Promise((c, e) => "my stubbed content entry");
     }
   
-    public async evaluateComponents() {
+    public async evaluateComponents(): Promise<any> {
       console.debug("evaluateComponents");
-      await this.performIqScan();
+      return await this.performIqScan();
     }
   
-    private async performIqScan() {
-      try {
-        let componentContainer = new ComponentContainer(this.requestService);
+    private async performIqScan(): Promise<any> {
+      return new Promise((resolve, reject) => {
+        try {
+          let componentContainer = new ComponentContainer(this.requestService);
 
-        let data: any;
+          window.withProgress(
+            {
+              location: ProgressLocation.Notification, 
+              title: "Running Nexus IQ Server Scan"
+            }, async (progress, token) => {
+              let data: any;
+              if (componentContainer.PackageMuncher != undefined) {
+                progress.report({message: "Starting to package your dependencies for IQ Server", increment: 5});
+                await componentContainer.PackageMuncher.packageForIq();
 
-        if (componentContainer.PackageMuncher != undefined) {
-          await componentContainer.PackageMuncher.packageForIq();
-  
-          data = await componentContainer.PackageMuncher.convertToNexusFormat();
-          this.components = componentContainer.PackageMuncher.toComponentEntries(data);
-          this.coordsToComponent = componentContainer.PackageMuncher.CoordinatesToComponents;
-        } else {
-          throw new TypeError("Unable to instantiate Package Muncher");
-        }
-
-        if (undefined == data) {
-          throw new RangeError("Attempted to generated dependency list but received an empty collection. NexusIQ will not be invoked for this project.");
-        }
-  
-        console.debug("getting applicationInternalId", this.applicationPublicId);
-        let response: string = await this.requestService.getApplicationId(this.applicationPublicId) as string;
-  
-        let appRep = JSON.parse(response);
-        console.debug("appRep", appRep);
-  
-        this.requestService.setApplicationId(appRep.applications[0].id)
-        console.debug("applicationInternalId", this.requestService.getApplicationInternalId());
-  
-        let resultId = await this.requestService.submitToIqForEvaluation(data, this.requestService.getApplicationInternalId());
-  
-        console.debug("report", resultId);
-        let resultDataString = await this.requestService.asyncPollForEvaluationResults(this.requestService.getApplicationInternalId(), resultId);
-        let resultData = JSON.parse(resultDataString as string);
-  
-        console.debug(`Received results from IQ scan:`, resultData);
-
-        for (let resultEntry of resultData.results) {
-          let componentEntry: ComponentEntry | undefined;
+                progress.report({message: "Reticulating splines...", increment: 25});
```

Comment:
![](https://66.media.tumblr.com/568161ab29c1e899ee423b4757b59d37/tumblr_mkr4sui0Rv1qdukbzo1_500.jpg)

Reply:
LOL glad you got got it